src/NServiceBus.Core/obsoletes-v9.cs

```diff
@@ -0,0 +1,82 @@
+namespace NServiceBus
+{
+    /// <summary>
+    /// Provides support for <see cref="UseTransport{T}"/> transport APIs.
+    /// </summary>
+    public static class LearningTransportApiExtensions
+    {
+        /// <summary>
+        /// Configures NServiceBus to use the given transport.
+        /// </summary>
+        [ObsoleteEx(
+            RemoveInVersion = "10",
+            TreatAsErrorFromVersion = "9",
+            ReplacementTypeOrMember = "EndpointConfiguration.UseTransport(TransportDefinition)")]
+        public static LearningTransportSettings UseTransport<T>(this EndpointConfiguration config)
+          where T : LearningTransport
+        {
+            var transport = new LearningTransport();
+
+            var routing = config.UseTransport(transport);
+
+            var settings = new LearningTransportSettings(transport, routing);
+
+            return settings;
+        }
+    }
+
+    /// <summary>
+    /// Learning transport configuration settings.
+    /// </summary>
+    public class LearningTransportSettings
```

Comment:
should there be some base class since they all need to support the `Routing()` method?

Reply:
Yeah, I think that base class makes sense. It would make sure we don't miss the extension method in the downstreams.

Reply:
Added the base class.